src/Test/Desktop/OpenRiaServices.Common.DomainServices.Test/Mocks/ServerSideAsyncDomainService.cs

```diff
@@ -97,6 +112,101 @@ public Task<RangeItem> GetRangeByIdAsync(int id)
                 );
         }
 
+        /// <summary>
+        /// Insert operation
+        /// </summary>
+        /// <param name="rangeItem">Item to insert</param>
+        /// <returns></returns>
+        [Insert]
+        public async Task InsertRangeAsync(RangeItem rangeItem)
+        {
+            await Delay(5);
+            rangeItem.Id = 42;
+        }
+
+        /// <summary>
+        /// Insert operation that throws exception in task
+        /// </summary>
+        /// <param name="rangeItem">Item to insert</param>
+        /// <returns></returns>
+        [Insert]
+        public async Task InsertRangeAsyncThrowsException(RangeItem2 rangeItem)
+        {
+            await Delay(5);
+            throw new DomainException(nameof(InsertRangeAsyncThrowsException), 25);
+        }
+
+        /// <summary>
+        /// Update operation that performs a short delay.
+        /// </summary>
+        /// <param name="rangeItem">Item to update</param>
+        /// <returns></returns>
+        [Update]
+        public async Task UpdateRangeAsync(RangeItem rangeItem)
+        {
+            await Delay(5);
+        }
+
+        /// <summary>
+        /// Async update operation that throws an exception in task
+        /// </summary>
+        /// <param name="rangeItem"></param>
+        /// <returns></returns>
+        [Update]
+        public async Task UpdateRangeAsyncThrowsException(RangeItem2 rangeItem)
+        {
+            await Delay(5);
+            throw new DomainException(nameof(UpdateRangeAsyncThrowsException), 26);
```

Comment:
Do you mean that all operations are executed even if one operation throws ValidationException? But nothing is submitted.

Reply:
I was mostly interested in seeing that the special code in `InvokeDomainOperationEntry` gets executed so that validationexceptions are still attached to the corresponding entity.

But it is a good idea :+1: to make sure to see that we can get validationerrors on 2 different entities in one submit . (The submit should still throw an exception and both entitites should have validation errors) that also covers the basic validation scenario so it is not necessary to have a separate validation test with only one entity.

Reply:
check!